src/server/scripts/Northrend/VioletHold/boss_ichoron.cpp

```diff
@@ -335,54 +441,72 @@ class npc_ichor_globule : public CreatureScript
     {
         npc_ichor_globuleAI(Creature* creature) : ScriptedAI(creature)
         {
-            Initialize();
             instance = creature->GetInstanceScript();
         }
 
-        void Initialize()
+        void Reset() override
         {
-            uiRangeCheck_Timer = 1000;
+            pathId = 0;
+            events.Reset();
+            DoCast(SPELL_WATER_GLOBULE);
         }
 
-        InstanceScript* instance;
-
-        uint32 uiRangeCheck_Timer;
-
-        void Reset() override
+        void AttackStart(Unit* /*who*/) override
         {
-            Initialize();
-            DoCast(me, SPELL_WATER_GLOBULE);
         }
 
-        void AttackStart(Unit* /*who*/) override
+        void SetData(uint32 id, uint32 data) override
         {
+            if (id == DATA_GLOBULE_PATH)
+            {
+                pathId = data;
+                me->GetMotionMaster()->MovePoint(0, globulePaths[pathId]);
+            }
         }
 
-        void UpdateAI(uint32 uiDiff) override
+        void MovementInform(uint32 type, uint32 id) override
         {
-            if (uiRangeCheck_Timer < uiDiff)
+            if (type != POINT_MOTION_TYPE)
+                return;
+
+            switch (id)
             {
-                if (Creature* ichoron = instance->GetCreature(DATA_ICHORON))
-                {
-                    if (me->IsWithinDist(ichoron, 2.0f, false))
-                    {
-                        if (ichoron->AI())
-                            ichoron->AI()->DoAction(ACTION_WATER_ELEMENT_HIT);
-                        me->DespawnOrUnsummon();
-                    }
-                }
-                uiRangeCheck_Timer = 1000;
+                case 0:
+                    me->GetMotionMaster()->MovementExpired();
+                    events.ScheduleEvent(EVENT_GLOBULE_MOVE, 500);
+                    break;
+                case 1:
+                    me->GetMotionMaster()->MovementExpired();
+                    if (Creature* ichoron = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_ICHORON)))
+                        me->GetMotionMaster()->MoveFollow(ichoron, 0.0f, 0.0f);
+                    break;
             }
-            else uiRangeCheck_Timer -= uiDiff;
         }
 
-        void JustDied(Unit* /*killer*/) override
+        // on retail spell casted on a creature's death are not casted after death but keeping mob at 1 health, casting it and then letting the mob die.
+        // this feature should be still implemented
+        void DamageTaken(Unit* attacker, uint32 &damage) override
```

Comment:
this didn't get fix yet https://travis-ci.org/TrinityCore/TrinityCore/builds/68491667#L1724 , even if https://github.com/TrinityCore/TrinityCore/issues/14952 was closed